python/PyQt/PyQt/common.py.in

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    common.py
+    ---------------------
+    Date                 : January 2024
+    Copyright            : (C) 2024 by Nyall Dawson
+    Email                : nyall dot dawson at gmail dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+import functools
+from enum import Enum
+
+def _handle_extended_enum_values(value):
+    """
+    If value is an Enum, returns an integer representation of its value wrapped
+    in the Enum class.
+    Otherwise returns value unchanged.
+    """
+    if isinstance(value, Enum):
+       if isinstance(value.value, Enum):
+          return value.__class__(int(value.value.value))
```

Comment:
A comment given an example of a situation when value.value can be an Enum could be helpful

Reply:
Sure, done!